src/core/session.js

```diff
@@ -138,6 +140,15 @@ export class Session {
     return this.history.restorationIdentifier
   }
 
+  get pageRefreshDebouncePeriod() {
+    return this.#pageRefreshDebouncePeriod
+  }
+
+  set pageRefreshDebouncePeriod(value) {
```

Comment:
If we can override refresh with the debounced version more than once, I think we need to grab the original version of the method in the constructor, so that we don't accumulate debounce delays on each assignment:

```js
// in the constructor
this.debouncedRefresh = this.refresh

// in the setter
this.refresh = debounce(this.debouncedRefresh.bind(this), value)
```

Reply:
Good catch! Done